lib/sidekiq/middleware/server/datadog.rb

```diff
@@ -29,12 +30,11 @@ def initialize(opts = {})
 
           @metric_name  = opts[:metric_name] || 'sidekiq.job'
           @statsd       = opts[:statsd] || ::Datadog::Statsd.new(statsd_host, statsd_port)
-          @tags         = opts[:tags] || []
-          @skip_tags    = Array(opts[:skip_tags]).map(&:to_s)
-
-          env  = Sidekiq.options[:environment] || ENV['RACK_ENV']
-          host = opts[:hostname] || ENV['INSTRUMENTATION_HOSTNAME'] || Socket.gethostname
-          setup_defaults(host: host, env: env)
+          @tag_builder  = Sidekiq::Datadog::TagBuilder.new(
```

Comment:
may I suggest to use `Sidekiq::Datadog::TagBuilder.new(**opts)` here and change the initializer to:

```
def initialize(custom_tags: [], skip_tags: [], custom_hostname: nil, **)
```